polkadot/node/subsystem-types/src/runtime_client.rs

```diff
@@ -18,17 +18,70 @@ use async_trait::async_trait;
 use polkadot_primitives::{
 	async_backing, runtime_api::ParachainHost, slashing, vstaging, Block, BlockNumber,
 	CandidateCommitments, CandidateEvent, CandidateHash, CommittedCandidateReceipt, CoreState,
-	DisputeState, ExecutorParams, GroupRotationInfo, Hash, Id, InboundDownwardMessage,
+	DisputeState, ExecutorParams, GroupRotationInfo, Hash, Header, Id, InboundDownwardMessage,
 	InboundHrmpMessage, OccupiedCoreAssumption, PersistedValidationData, PvfCheckStatement,
 	ScrapedOnChainVotes, SessionIndex, SessionInfo, ValidationCode, ValidationCodeHash,
 	ValidatorId, ValidatorIndex, ValidatorSignature,
 };
+use sc_client_api::HeaderBackend;
 use sc_transaction_pool_api::OffchainTransactionPoolFactory;
-use sp_api::{ApiError, ApiExt, ProvideRuntimeApi};
+use sp_api::{ApiError, ApiExt, HeaderT, NumberFor, ProvideRuntimeApi};
 use sp_authority_discovery::AuthorityDiscoveryApi;
+use sp_blockchain::Info;
 use sp_consensus_babe::{BabeApi, Epoch};
 use std::{collections::BTreeMap, sync::Arc};
 
+/// Offers header utilities.
+///
+/// This is a async wrapper trait for ['HeaderBackend'] to be used with the
+/// `ChainApiSubsystem`.
+// This trait was introduced to suit the needs of collators. Depending on their operating mode, they
+// might not have a client of the relay chain that can supply a synchronous HeaderBackend
+// implementation.
+#[async_trait]
+pub trait ChainApiBackend: Send + Sync {
+	/// Get block header. Returns `None` if block is not found.
+	async fn header(&self, hash: Hash) -> sp_blockchain::Result<Option<Header>>;
+	/// Get blockchain info.
+	async fn info(&self) -> sp_blockchain::Result<Info<Block>>;
```

Comment:
It seems at the present, we don't need the full info to satisfy the request, just the latest finalized number. 
If we return just that, the implementation for collator side can be simplified. That being said, collator should not be (transitively) calling this method AFAIK (at least minimal relay chain node), so it should be fine as is.